red/runtime/nodes/flows/Flow.js

```diff
@@ -118,10 +120,28 @@ function Flow(global,flow) {
                 } else if (node.type === "status") {
                     statusNodeMap[node.z] = statusNodeMap[node.z] || [];
                     statusNodeMap[node.z].push(node);
+                } else if (node.type === "success") {
+                    succNodeMap[node.z] = succNodeMap[node.z] || [];
+                    succNodeMap[node.z].push(node);
                 }
             }
         }
-    }
+        for (id in activeNodes) {
```

Comment:
Not convinced this is necessary. Let the nodes report success and the flow can no-op the request if needed.

Reply:
This is for optimization of critical path of node execution as described in the above comment.  
A flow can be executed without this, but needs checking of existence of corresponding `Success` (or `Done`)  node.